pogo/demo.py

```diff
@@ -165,32 +165,27 @@ def getInventory(session):
 # Since traveling salesman problem, not
 # true solution. But at least you get
 # those step in
-def sortCloseForts(session):
-    # Sort nearest forts (pokestop)
+def sortCloseForts(session, type=1):
+    # Sort nearest forts (pokestop type 1, gyms type =)
     logging.info("Sorting Nearest Forts:")
     cells = session.getMapObjects()
     latitude, longitude, _ = session.getCoordinates()
-    ordered_forts = []
-    for cell in cells.map_cells:
-        for fort in cell.forts:
-            dist = Location.getDistance(
-                latitude,
-                longitude,
-                fort.latitude,
-                fort.longitude
-            )
-            if fort.type == 1:
-                ordered_forts.append({'distance': dist, 'fort': fort})
+    forts = [fort for cell in cells.map_cells for fort in cell.forts if fort.type == type]
+    return sorted(forts, key=lambda f: Location.getDistance(latitude, longitude, f.latitude, f.longitude))
+
 
-    ordered_forts = sorted(ordered_forts, key=lambda k: k['distance'])
-    return [instance['fort'] for instance in ordered_forts]
+# Find the gym closest to user
+def findClosestGym(session):
+    # Find nearest fort (gyms)
+    logging.info("Finding Nearest Gym:")
+    return sortCloseForts(session, 0)[0] if sortCloseForts(session) else None
 
 
 # Find the fort closest to user
 def findClosestFort(session):
     # Find nearest fort (pokestop)
     logging.info("Finding Nearest Fort:")
-    return sortCloseForts(session)[0]
+    return sortCloseForts(session)[0] if sortCloseForts(session) else None
 
```

Comment:
Notice how that would call sortCloseForts(session) twice, invoking multiple requests. Set forts to a temp variable instead.